app/screens/theme/theme.js

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View, FlatList} from 'react-native';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+
+import StatusBar from 'app/components/status_bar';
+import Section from 'app/screens/settings/section';
+import SectionItem from 'app/screens/settings/section_item';
+import FormattedText from 'app/components/formatted_text';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import Preferences from 'mattermost-redux/constants/preferences';
+
+export default class Theme extends React.PureComponent {
+    static propTypes = {
+        teamId: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        userId: PropTypes.string.isRequired,
+        actions: PropTypes.shape({
+            savePreferences: PropTypes.func.isRequired,
+        }).isRequired,
+        allowedThemes: PropTypes.arrayOf(PropTypes.object),
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    setTheme = (key) => {
+        const {userId, teamId, actions: {savePreferences}, allowedThemes} = this.props;
+        const selectedTheme = allowedThemes.find((theme) => theme.key === key);
+
+        savePreferences(userId, [{
+            user_id: userId,
+            category: Preferences.CATEGORY_THEME,
+            name: teamId,
+            value: JSON.stringify(selectedTheme),
+        }]);
+    }
+
+    renderThemeRow = ({item}) => {
+        const {theme} = this.props;
+        const style = getStyleSheet(theme);
+
+        return (
+            <View>
+                <SectionItem
+                    label={(
+                        <FormattedText
+                            id={`user.settings.display.${item.type}`}
+                            defaultMessage={item.type}
+                        />
+                    )}
+                    action={this.setTheme}
+                    actionType='select'
+                    actionValue={item.key}
+                    selected={item.type.toLowerCase() === theme.type.toLowerCase()}
+                    theme={theme}
+                />
+                <View style={style.divider}/>
+            </View>
+        );
+    };
+
+    render() {
+        const {theme, allowedThemes} = this.props;
+        const style = getStyleSheet(theme);
+        return (
+            <View style={style.container}>
+                <StatusBar/>
+                <View style={style.wrapper}>
+                    <Section
+                        disableHeader={true}
+                        theme={theme}
+                    >
+                        <FlatList
+                            data={allowedThemes}
+                            renderItem={this.renderThemeRow}
+                            keyExtractor={(item) => item.key}
```

Comment:
Can you make this a method on the `Theme` component? It's a good practice to avoid arrow functions whenever possible to prevent child components from re-rendering because they've received a new copy of the function as a prop